### PR TITLE
PMP: Introduce `stitch_boundary_cycle`

### PR DESCRIPTION
**Pull request by MaelRL**
This PR introduces a function `stitch_boundary_cycle()` which only stitches within a prescribed border. This is mostly extracting code from `stitch_boundary_cycle**s**()` + some interfacing.

TODO:
- [x] Document both `stitch_boundary_cycle(s)` ? (@sloriot)
- [x] Changes.md

## Release Management

* Affected package(s): `PMP`
* Feature/Small Feature (if any): [Stitch_boundary_cycle](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Stitch_boundary_cycle)
Pre-approved --[[User:Sloriot|Sloriot]] 13:38, 18 April 2019 (CEST)

### CODE REVIEW

**Comment by sloriot** [comment]
I think it makes sense to document both.

**Comment by MaelRL** [comment]
See above for the small feature link.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/stitch_borders.h [hunk 1]
```diff
@@ -529,25 +529,36 @@ void stitch_borders_impl(PM& pmesh,
...
-//   \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`.
-//     If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` must be available in `PolygonMesh`.
-//   \cgalParamEnd
-// \cgalNamedParamsEnd
-//
+} //end of namespace internal
+
+/// \ingroup PMP_repairing_grp
+///
+/// Stitches together, whenever possible, two halfedges of the border incident to halfedge `h`.
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/stitch_borders.h]:
```suggestion
/// Stitches together, whenever possible, two halfedges belonging to the boundary cycle described by the halfedge `h`.
```

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/stitch_borders.h [hunk 2]
```diff
@@ -573,70 +581,129 @@ std::size_t stitch_boundary_cycles(PolygonMesh& pm,
...
+template <typename PolygonMesh>
+std::size_t stitch_boundary_cycle(const typename boost::graph_traits<PolygonMesh>::halfedge_descriptor h,
+                                  PolygonMesh& pm)
+{
+  return stitch_boundary_cycle(h, pm, CGAL::parameters::all_default());
+}
+
+/// \ingroup PMP_repairing_grp
+///
+/// Stitches together, whenever possible, two halfedges living on the same border.
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/stitch_borders.h]:
```suggestion
/// Stitches together, whenever possible, two halfedges belonging to the same boundary cycle.
```

**Comment by MaelRL** [comment]
Merge conflicts really ought to be shown in the PR list, it's really tedious having to check individually.

**Comment by lrineau** [comment]
[**@MaelRL**](https://github.com/MaelRL) commented on [May 20, 2019, 12:07 PM GMT+2](https://github.com/CGAL/cgal/pull/3861#issuecomment-493922735 "2019-05-20T10:07:40Z - Replied by Github Reply Comments"):
> Merge conflicts really ought to be shown in the PR list, it's really tedious having to check individually.

I agree. But that is a RFE to fill to Github. Actually, that is even worse: Github tests for conflicts lazily: after `master` has been updated, the merge is only tested the first time somebody goes to the PR web page. A bot could try that: after a change to releases branches or master, visit all pull-requests, and then check for conflicts.

**Comment by MaelRL** [comment]
I know it's on Github's side. I saw that there was [mergeable](https://developer.github.com/v4/object/pullrequest/#mergeable) flag in the PR API (but would it count as a query that updates the flag after a master merge?), but I don't know if I'm annoyed enough to write an extension :).

**Comment by lrineau** [comment]
Doxygen warning:
> include/CGAL/Polygon_mesh_processing/stitch_borders.h:559: warning: The following parameters of CGAL::Polygon_mesh_processing::stitch_boundary_cycle(const typename boost::graph_traits< PolygonMesh >::halfedge_descriptor h, PolygonMesh &pm, const NamedParameters &np) are not documented:
  parameter 'h'

https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.0-Ic-53/logs_master/Polygon_mesh_processing.log

**Comment by maxGimeno** [comment]
Seems all good in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-59.shtml

**Pull request merged**